osh/cmd_parse.py

```diff
@@ -584,14 +585,21 @@ def _Eat(self, c_id, msg=None):
     self._Next()
     return skipped
 
-  def _NewlineOk(self):
+  def _NewlineOkLegacy(self):
     # type: () -> None
     """Check for optional newline and consume it."""
     self._Peek()
     if self.c_id == Id.Op_Newline:
       self._Next()
       self._Peek()
 
+  def _NewlineOk(self):
+    # type: () -> None
+    """Check for optional newline and consume it."""
+    if self.w_parser.LookPastSpace() == Id.Op_Newline:
+      self._Peek()
+      self._Next()
```

Comment:
Hm I would have to go back to my branch (dev-andy-vb8), but I was leaning toward keeping this function like the old one, but removing the last _Next()

I thought that worked, but again we didn't test every case